src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -156,8 +158,27 @@ public void rm(@Nonnull EnvVars launchEnv, @Nonnull String containerId) throws I
             return null;
         }
     }
-
-    private Date getCreatedDate(@Nonnull EnvVars launchEnv, @Nonnull String objectId) throws IOException, InterruptedException {
+    
+    /**
+     * Inspect a docker image/container.
+     * @param launchEnv Docker client launch environment.
+     * @param objectId The image/container ID.
+     * @param fieldPath The data path of the data required e.g. {@code .NetworkSettings.IPAddress}.
+     * @return The inspected field value. May be an empty string
+     * @throws IOException Execution error. Also fails if cannot retrieve the requested field from the request
+     * @throws InterruptedException Interrupted
+     * @since TODO
+     */
+    public @Nonnull String inspectRequiredField(@Nonnull EnvVars launchEnv, @Nonnull String objectId, 
+            @Nonnull String fieldPath) throws IOException, InterruptedException {
+        final String fieldValue = inspect(launchEnv, objectId, fieldPath);
+        if (fieldValue == null) {
+            throw new IOException("Cannot retrieve " + fieldPath + " for id=" +objectId);
```

Comment:
I think `IOException` is fine here. Does not really matter, since the detail is in the message.